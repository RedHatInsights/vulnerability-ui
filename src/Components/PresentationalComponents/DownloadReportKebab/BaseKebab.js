/* eslint-disable camelcase */
import { Dropdown, KebabToggle } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React, { useState } from 'react';

const BaseKebab = ({ dropdownItems, disabled, ...props }) => {
    const [isOpen, setOpen] = useState(false);

    function handleOnKebabToggle(kebabState) {
        setOpen(kebabState);
    }

    function handleOnKebabSelect() {
        setOpen(!isOpen);
    }

    return (
        <div>
            <Dropdown
                onSelect={handleOnKebabSelect}
                toggle={<KebabToggle onToggle={handleOnKebabToggle} isDisabled={disabled} />}
                isOpen={isOpen}
                isPlain
                dropdownItems={dropdownItems}
                {...props}
            />
        </div>
    );
};

BaseKebab.propTypes = {
    dropdownItems: propTypes.array,
    disabled: propTypes.bool
};

export default BaseKebab;
