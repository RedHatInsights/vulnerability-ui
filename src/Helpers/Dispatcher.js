import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        if ('json' in res) {
            let parsedMessage = res.json();
            parsedMessage.then(arr => {
                for (let err of arr.errors) {
                    store.dispatch(
                        addNotification({
                            variant: 'danger',
                            title: 'Error ' + err.status,
                            description: err.detail
                        })
                    );
                }
            });
        } else {
            store.dispatch(
                addNotification({
                    variant: 'danger',
                    title: 'Error ' + res.status,
                    description: res.statusText
                })
            );
        }
    });
}
