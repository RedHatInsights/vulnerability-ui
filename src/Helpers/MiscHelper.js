import { SecurityIcon, UnknownIcon } from '@patternfly/react-icons';
import { SortByDirection } from '@patternfly/react-table';
import findIndex from 'lodash/findIndex';
import propTypes from 'prop-types';
import React from 'react';
import { impactColorList, impactList, PUBLIC_DATE_OPTIONS } from './constants';
import qs from 'query-string';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});

/**
 * Based on the  allowedParams it will construct the arguments for the API call
 * which is defined in the vulnerabilities-client. vulnerabilities-client is written is typescript
 * so we need to define and pass every single parameter. Order in allowedParams is important
 *
 * @param {Object} apiProps
 * @param {Array} allowedParams - order should align with the params in the vulnerabilities-client
 */
export function constructParameters(apiProps, allowedParams) {
    if (apiProps) {
        Object.keys(apiProps).forEach(
            key => (
                apiProps[key] === undefined
                || apiProps[key] === ''
            )
            && delete apiProps[key]
        );

        const params = allowedParams.map(item => apiProps[item]);

        return params;
    }

    return [];
}

// TODO DRY:similar to constructParameters
export function constructURLParameters(urlParams, allowedParams) {
    if (urlParams) {
        const params = { ...urlParams };
        Object.keys(urlParams).forEach(
            key => (
                params[key] === undefined
                || params[key] === ''
                || !allowedParams.includes(key)
                || params[key] === false
            )
            && delete params[key]

        );
        return params;
    }
}

export function formatDate(date = new Date(), includeTime = false) {
    const prepend = (number) => `${`${number}`.length === 1 ? '0' : ''}${number}`;
    const toFormat = new Date(date);
    if (toFormat instanceof Date && !isNaN(toFormat)) {
        const year = toFormat.getFullYear();
        const month = toFormat.getMonth() + 1; // month is zero indexed
        const day = toFormat.getDate();
        const hour = toFormat.getUTCHours();
        const minute = toFormat.getUTCMinutes();

        return includeTime
            ? `${year}-${prepend(month)}-${prepend(day)} ${prepend(hour)}:${prepend(minute)} UTC`
            : `${year}-${prepend(month)}-${prepend(day)}`;
    }

    return toFormat;
}

/**
 * Contructs the filters which have more than one parameter that needs to be passed
 * and resets all the parameteres when it's needed
 *
 * @param {Object} filterParams
 * @return {Object}
 */
export function constructFilterParameters(filterParams) {
    if (Object.prototype.hasOwnProperty.call(filterParams, 'publish_date')) {
        let publicEntry = PUBLIC_DATE_OPTIONS.find(item => item.value === filterParams.publish_date);

        if (typeof publicEntry !== 'undefined') {
            filterParams.public_from = publicEntry.from && formatDate(publicEntry.from);
            filterParams.public_to = publicEntry.to && formatDate(publicEntry.to);
        } else {
            filterParams.public_from = undefined;
            filterParams.public_to = undefined;
            filterParams.publish_date = undefined;
        }
    }

    return filterParams;
}

export function getImpactDetails(impact) {
    if (Object.keys(impactList).includes(impact)) {
        return {
            title: impactList[impact].title,
            color: impactList[impact].color,
            text: impactList[impact].text,
            titleContent: (
                <SecurityIcon size="lg" />
            )
        };
    } else {
        return {
            title: 'Unknown',
            color: impactColorList.unknown,
            text: '',
            titleContent: (
                <UnknownIcon size="lg" />
            )
        };
    }
}

export const createSortBy = (header, value) => {
    if (value) {
        let direction = value.indexOf('-') === 0 ? SortByDirection.desc : SortByDirection.asc;
        value = value.replace(/^(-|\+)/, '');
        const index = findIndex(header, item => item.key === value);
        let sort = {
            index,
            direction
        };
        return sort;
    }

    return {};
};

export const handleSortColumn = (key, direction, header, currentSort, apply) => {
    let columnName = header[key].key;
    const useDefault = currentSort && currentSort.substr(1) !== columnName;
    if (direction === SortByDirection.desc || useDefault) {
        columnName = '-' + columnName;
    }

    apply({ sort: columnName });
};

export const updateStateSet = (stateSet, names, value) => {
    [].concat(names).forEach(name => {
        value ? stateSet.add(name) : stateSet.delete(name);
    });
    return stateSet;
};

export const useUrlParams = (allowedParams) => {
    const url = new URL(window.location);
    const urlParams = qs.parse(url.search);

    const setUrlParams = (parameters) => {
        const searchParams = qs.stringify(constructURLParameters(parameters, allowedParams));
        window.history.replaceState(null, null, `${url.origin}${url.pathname}?${searchParams}`);
    };

    return [urlParams, setUrlParams];
};

export const updateRef = (meta, apply) => {
    const pages = parseInt(meta.pages);
    const page = parseInt(meta.page);
    const cvesCount = parseInt(meta.cvesCount);
    const goTo = (pages === page && page > 1 && cvesCount === 1) ? (pages - 1) : page;

    apply({ page: goTo });
};

export const mountWithIntl = (Component) => {
    const wrapper = mount(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const shallowWithIntl = (Component) => {
    const wrapper = shallow(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const insertIf = (condition, ...elements) => {
    return condition ? elements : [];
};
