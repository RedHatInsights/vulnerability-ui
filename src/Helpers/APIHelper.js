let BASE_ROUTE = '/api/vulnerability/v1';
import { GitApi } from '@redhat-cloud-services/vulnerabilities-client';
import { SystemProfileApi } from '@redhat-cloud-services/host-inventory-client';
import instance from '../Utilities/interceptors';
import { constructParameters } from './MiscHelper';

let api = new GitApi(undefined, BASE_ROUTE, instance);

const INVENTORY_API_BASE = '/api/inventory/v1';
const systemProfile = new SystemProfileApi(undefined, INVENTORY_API_BASE, instance);

const defaultParams = [
    'filter',
    'limit',
    'offset',
    'page',
    'page_size',
    'sort'
];

export function getCveDetails(synopsis) {
    const result = api.getCveDetails(synopsis);
    return result;
}

export function getSystemDetails(system) {
    const result = api.getSystemDetails(system);
    return result;
}

export function setSystemCveStatus(params) {
    let result = api.setStatus(params);
    return result;
}

export function optOutSystems(systems, optOut) {
    const data = { opt_out: optOut, inventory_id: new Array(...systems) };
    let result = api.setSystemsOptOut(data);
    return result;
}

export function setBusinessRisk(params) {
    let result = api.setCveBusinessRisk(params);
    return result;
}

export function setCveStatus(params) {
    let result = api.setCveStatus(params);
    return result;
}

/**
 * /vulnerabilities/cves
 * /vulnerabilities/cves/ids
 */
let cveParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'data_format',
    'business_risk_id',
    'status_id',
    'rule_presence',
    'tags',
    'sap_sids',
    'sap_system',
    'known_exploit',
    'affecting',
    'rhel_version'
];

export function getCveListByAccount(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'report',
        'advanced_report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveList(...parameterArray);
    return result;
}

export function getCveIdsList(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveIdsList(...parameterArray);
    return result;
}

/**
 * /cves/ID/affected_systems
 */
let systemsByCVEparams = [
    'status_id',
    'data_format',
    'uuid',
    'rule_key',
    'rule_presence',
    'rule',
    'tags',
    'sap_sids',
    'sap_system',
    'show_advisories',
    'advisory',
    'rhel_version',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getAffectedSystemsByCVE({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'report',
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsByCve(id, ...parameterArray);
    return result;
}

export function getAffectedSystemsIdsByCve({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsIdsByCve(id, ...parameterArray);
    return result;
}

/**
 * /systems/ID/cves
 * /systems/ID/cves/ids
 */
let cveBySystemParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'status_id',
    'data_format',
    'business_risk_id',
    'rule_presence',
    'show_advisories',
    'advisory',
    'rule_key',
    'known_exploit',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getCveListBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams,
        'report'
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveListBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

export function getCveIdsBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveIdsBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

/**
 * /systems/ids
 * /systems
 */
let systemsParams = [
    'data_format',
    'stale',
    'uuid',
    'tags',
    'sap_sids',
    'sap_system',
    'excluded',
    'rhel_version'
];

export function getSystems(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams,
        'report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsList(...parameterArray);
    return result;
}

export function getSystemsIds(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsIds(...parameterArray);
    return result;
}

export function getAnnouncement() {
    let result = api.getAnnouncement();
    return result;
}

export function getDashbar(apiProps) {
    let parameterNames = [
        'tags',
        'sap_sids',
        'sap_system',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getDashbar(...parameterArray);
    return result;
}

export function getOperatingSystems() {
    return systemProfile.apiSystemProfileGetOperatingSystem();
}

export function fetchPdfReport(template, filters) {
    const CRC_PDF_GENERATE_API = '/api/crc-pdf-generator/v1/generate';
    const url = new URL(CRC_PDF_GENERATE_API,  window.location.origin);

    return fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },

        body: JSON.stringify({
            service: 'vulnerability',
            template,
            params: {
                ...filters
            }
        })
    }, 50000)
        .then((response) => response.blob());
}
