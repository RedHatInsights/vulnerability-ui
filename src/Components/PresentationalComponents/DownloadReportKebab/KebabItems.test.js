import React from 'react';
import {
    kebabItemDownloadJSON,
    kebabItemDownloadCSV,
    kebabItemEditBusinessRisk,
    kebabItemEditStatus
} from './KebabItems';
import toJson from 'enzyme-to-json';
import BaseKebab from './BaseKebab';
import { mountWithIntl } from '../../../Helpers/MiscHelper';

describe('KebabItems component', () => {
    it('Should render without default params', () => {
        const option1 = (function1) => mountWithIntl(
            <DropdownItem key="o1" component="button" onClick={() => function1()}>
                Option1
            </DropdownItem>
        );

        const items = [
            kebabItemDownloadJSON(),
            kebabItemDownloadCSV(),
            kebabItemEditBusinessRisk(),
            kebabItemEditStatus()
        ]

        const wrapper = mountWithIntl(<BaseKebab dropdownItems={items}/>);
        
        const kebabButton = wrapper.find('button');
        kebabButton.simulate('click');

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
