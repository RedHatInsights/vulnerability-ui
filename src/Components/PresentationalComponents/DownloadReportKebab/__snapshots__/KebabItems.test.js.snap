// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KebabItems component Should render without default params 1`] = `
<BaseKebab
  dropdownItems={
    Array [
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Export as JSON"
          description="Kebab item"
          id="kebab.exportAsJson"
        />
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Export as CSV"
          description="Kebab item"
          id="kebab.exportAsCsv"
        />
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Edit business risk"
          description="Kebab item"
          id="kebab.editBusinnesRisk"
        />
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Edit status"
          description="Kebab item"
          id="kebab.editStatus"
        />
      </DropdownItem>,
    ]
  }
>
  <div>
    <Dropdown
      dropdownItems={
        Array [
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Export as JSON"
              description="Kebab item"
              id="kebab.exportAsJson"
            />
          </DropdownItem>,
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Export as CSV"
              description="Kebab item"
              id="kebab.exportAsCsv"
            />
          </DropdownItem>,
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit business risk"
              description="Kebab item"
              id="kebab.editBusinnesRisk"
            />
          </DropdownItem>,
          <DropdownItem
            component="button"
            onClick={[Function]}
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Edit status"
              description="Kebab item"
              id="kebab.editStatus"
            />
          </DropdownItem>,
        ]
      }
      isOpen={true}
      isPlain={true}
      onSelect={[Function]}
      toggle={
        <KebabToggle
          onToggle={[Function]}
        />
      }
    >
      <DropdownWithContext
        autoFocus={true}
        className=""
        direction="down"
        dropdownItems={
          Array [
            <DropdownItem
              component="button"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Export as JSON"
                description="Kebab item"
                id="kebab.exportAsJson"
              />
            </DropdownItem>,
            <DropdownItem
              component="button"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Export as CSV"
                description="Kebab item"
                id="kebab.exportAsCsv"
              />
            </DropdownItem>,
            <DropdownItem
              component="button"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Edit business risk"
                description="Kebab item"
                id="kebab.editBusinnesRisk"
              />
            </DropdownItem>,
            <DropdownItem
              component="button"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Edit status"
                description="Kebab item"
                id="kebab.editStatus"
              />
            </DropdownItem>,
          ]
        }
        isGrouped={false}
        isOpen={true}
        isPlain={true}
        menuAppendTo="inline"
        onSelect={[Function]}
        position="left"
        toggle={
          <KebabToggle
            onToggle={[Function]}
          />
        }
      >
        <div
          className="pf-c-dropdown pf-m-expanded"
          data-ouia-component-id="OUIA-Generated-Dropdown-1"
          data-ouia-component-type="PF4/Dropdown"
          data-ouia-safe={true}
        >
          <KebabToggle
            aria-haspopup={true}
            getMenuRef={[Function]}
            id="pf-dropdown-toggle-id-2"
            isOpen={true}
            isPlain={true}
            key=".0"
            onEnter={[Function]}
            onToggle={[Function]}
            parentRef={
              Object {
                "current": <div
                  class="pf-c-dropdown pf-m-expanded"
                  data-ouia-component-id="OUIA-Generated-Dropdown-1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="true"
                    aria-haspopup="true"
                    aria-label="Actions"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    id="pf-dropdown-toggle-id-2"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 192 512"
                      width="1em"
                    >
                      <path
                        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      />
                    </svg>
                  </button>
                  <ul
                    aria-labelledby="pf-dropdown-toggle-id-2"
                    class="pf-c-dropdown__menu"
                    role="menu"
                  >
                    <li
                      role="menuitem"
                    >
                      <button
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        tabindex="-1"
                        type="button"
                      >
                        Export as JSON
                      </button>
                    </li>
                    <li
                      role="menuitem"
                    >
                      <button
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        tabindex="-1"
                        type="button"
                      >
                        Export as CSV
                      </button>
                    </li>
                    <li
                      role="menuitem"
                    >
                      <button
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        tabindex="-1"
                        type="button"
                      >
                        Edit business risk
                      </button>
                    </li>
                    <li
                      role="menuitem"
                    >
                      <button
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        tabindex="-1"
                        type="button"
                      >
                        Edit status
                      </button>
                    </li>
                  </ul>
                </div>,
              }
            }
          >
            <Toggle
              aria-haspopup={true}
              aria-label="Actions"
              bubbleEvent={false}
              className=""
              getMenuRef={[Function]}
              id="pf-dropdown-toggle-id-2"
              isActive={false}
              isDisabled={false}
              isOpen={true}
              isPlain={true}
              isPrimary={false}
              isSplitButton={false}
              onEnter={[Function]}
              onToggle={[Function]}
              parentRef={
                Object {
                  "current": <div
                    class="pf-c-dropdown pf-m-expanded"
                    data-ouia-component-id="OUIA-Generated-Dropdown-1"
                    data-ouia-component-type="PF4/Dropdown"
                    data-ouia-safe="true"
                  >
                    <button
                      aria-expanded="true"
                      aria-haspopup="true"
                      aria-label="Actions"
                      class="pf-c-dropdown__toggle pf-m-plain"
                      id="pf-dropdown-toggle-id-2"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 192 512"
                        width="1em"
                      >
                        <path
                          d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                        />
                      </svg>
                    </button>
                    <ul
                      aria-labelledby="pf-dropdown-toggle-id-2"
                      class="pf-c-dropdown__menu"
                      role="menu"
                    >
                      <li
                        role="menuitem"
                      >
                        <button
                          aria-disabled="false"
                          class="pf-c-dropdown__menu-item"
                          data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                          data-ouia-component-type="PF4/DropdownItem"
                          data-ouia-safe="true"
                          tabindex="-1"
                          type="button"
                        >
                          Export as JSON
                        </button>
                      </li>
                      <li
                        role="menuitem"
                      >
                        <button
                          aria-disabled="false"
                          class="pf-c-dropdown__menu-item"
                          data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                          data-ouia-component-type="PF4/DropdownItem"
                          data-ouia-safe="true"
                          tabindex="-1"
                          type="button"
                        >
                          Export as CSV
                        </button>
                      </li>
                      <li
                        role="menuitem"
                      >
                        <button
                          aria-disabled="false"
                          class="pf-c-dropdown__menu-item"
                          data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                          data-ouia-component-type="PF4/DropdownItem"
                          data-ouia-safe="true"
                          tabindex="-1"
                          type="button"
                        >
                          Edit business risk
                        </button>
                      </li>
                      <li
                        role="menuitem"
                      >
                        <button
                          aria-disabled="false"
                          class="pf-c-dropdown__menu-item"
                          data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                          data-ouia-component-type="PF4/DropdownItem"
                          data-ouia-safe="true"
                          tabindex="-1"
                          type="button"
                        >
                          Edit status
                        </button>
                      </li>
                    </ul>
                  </div>,
                }
              }
            >
              <button
                aria-expanded={true}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={false}
                id="pf-dropdown-toggle-id-2"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <EllipsisVIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 192 512"
                    width="1em"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                    />
                  </svg>
                </EllipsisVIcon>
              </button>
            </Toggle>
          </KebabToggle>
          <DropdownMenu
            aria-labelledby="pf-dropdown-toggle-id-2"
            autoFocus={false}
            className=""
            component="ul"
            isGrouped={false}
            isOpen={true}
            openedOnEnter={false}
            position="left"
            setMenuComponentRef={[Function]}
          >
            <ul
              aria-labelledby="pf-dropdown-toggle-id-2"
              autoFocus={false}
              className="pf-c-dropdown__menu"
              hidden={false}
              role="menu"
            >
              <DropdownItem
                component="button"
                index={0}
                key=".$json"
                onClick={[Function]}
              >
                <InternalDropdownItem
                  className=""
                  component="button"
                  context={
                    Object {
                      "keyHandler": [Function],
                      "sendRef": [Function],
                    }
                  }
                  data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                  data-ouia-component-type="PF4/DropdownItem"
                  data-ouia-safe={true}
                  description={null}
                  enterTriggersArrowDown={false}
                  icon={null}
                  index={0}
                  inoperableEvents={
                    Array [
                      "onClick",
                      "onKeyPress",
                    ]
                  }
                  isAriaDisabled={false}
                  isDisabled={false}
                  isHovered={false}
                  isPlainText={false}
                  onClick={[Function]}
                  role="menuitem"
                  styleChildren={true}
                  tabIndex={-1}
                  tooltipProps={Object {}}
                >
                  <li
                    className={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                  >
                    <button
                      aria-disabled={false}
                      className="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe={true}
                      tabIndex={-1}
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="Export as JSON"
                        description="Kebab item"
                        id="kebab.exportAsJson"
                      >
                        Export as JSON
                      </FormattedMessage>
                    </button>
                  </li>
                </InternalDropdownItem>
              </DropdownItem>
              <DropdownItem
                component="button"
                index={1}
                key=".$csv"
                onClick={[Function]}
              >
                <InternalDropdownItem
                  className=""
                  component="button"
                  context={
                    Object {
                      "keyHandler": [Function],
                      "sendRef": [Function],
                    }
                  }
                  data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                  data-ouia-component-type="PF4/DropdownItem"
                  data-ouia-safe={true}
                  description={null}
                  enterTriggersArrowDown={false}
                  icon={null}
                  index={1}
                  inoperableEvents={
                    Array [
                      "onClick",
                      "onKeyPress",
                    ]
                  }
                  isAriaDisabled={false}
                  isDisabled={false}
                  isHovered={false}
                  isPlainText={false}
                  onClick={[Function]}
                  role="menuitem"
                  styleChildren={true}
                  tabIndex={-1}
                  tooltipProps={Object {}}
                >
                  <li
                    className={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                  >
                    <button
                      aria-disabled={false}
                      className="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe={true}
                      tabIndex={-1}
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="Export as CSV"
                        description="Kebab item"
                        id="kebab.exportAsCsv"
                      >
                        Export as CSV
                      </FormattedMessage>
                    </button>
                  </li>
                </InternalDropdownItem>
              </DropdownItem>
              <DropdownItem
                component="button"
                index={2}
                key=".$editBusinessRisk"
                onClick={[Function]}
              >
                <InternalDropdownItem
                  className=""
                  component="button"
                  context={
                    Object {
                      "keyHandler": [Function],
                      "sendRef": [Function],
                    }
                  }
                  data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                  data-ouia-component-type="PF4/DropdownItem"
                  data-ouia-safe={true}
                  description={null}
                  enterTriggersArrowDown={false}
                  icon={null}
                  index={2}
                  inoperableEvents={
                    Array [
                      "onClick",
                      "onKeyPress",
                    ]
                  }
                  isAriaDisabled={false}
                  isDisabled={false}
                  isHovered={false}
                  isPlainText={false}
                  onClick={[Function]}
                  role="menuitem"
                  styleChildren={true}
                  tabIndex={-1}
                  tooltipProps={Object {}}
                >
                  <li
                    className={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                  >
                    <button
                      aria-disabled={false}
                      className="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe={true}
                      tabIndex={-1}
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="Edit business risk"
                        description="Kebab item"
                        id="kebab.editBusinnesRisk"
                      >
                        Edit business risk
                      </FormattedMessage>
                    </button>
                  </li>
                </InternalDropdownItem>
              </DropdownItem>
              <DropdownItem
                component="button"
                index={3}
                key=".$editStatus"
                onClick={[Function]}
              >
                <InternalDropdownItem
                  className=""
                  component="button"
                  context={
                    Object {
                      "keyHandler": [Function],
                      "sendRef": [Function],
                    }
                  }
                  data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                  data-ouia-component-type="PF4/DropdownItem"
                  data-ouia-safe={true}
                  description={null}
                  enterTriggersArrowDown={false}
                  icon={null}
                  index={3}
                  inoperableEvents={
                    Array [
                      "onClick",
                      "onKeyPress",
                    ]
                  }
                  isAriaDisabled={false}
                  isDisabled={false}
                  isHovered={false}
                  isPlainText={false}
                  onClick={[Function]}
                  role="menuitem"
                  styleChildren={true}
                  tabIndex={-1}
                  tooltipProps={Object {}}
                >
                  <li
                    className={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                  >
                    <button
                      aria-disabled={false}
                      className="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe={true}
                      tabIndex={-1}
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="Edit status"
                        description="Kebab item"
                        id="kebab.editStatus"
                      >
                        Edit status
                      </FormattedMessage>
                    </button>
                  </li>
                </InternalDropdownItem>
              </DropdownItem>
            </ul>
          </DropdownMenu>
        </div>
      </DropdownWithContext>
    </Dropdown>
  </div>
</BaseKebab>
`;
